src/main/java/uk/gov/gchq/magmacore/demo/ExampleAssociations.java

```diff
@@ -34,53 +39,46 @@
-            final FunctionalSystem house,
-            final IRI beginning,
+    private static void occupyHouse(final MagmaCoreService mcService, final List<DbCreateOperation> creates,
+            final PossibleWorld possibleWorld, final Person person, final FunctionalSystem house, final IRI beginning,
             final IRI ending) {
 
-        final var entities = mcService.findByEntityNameInTransaction(List.of(
-                    "CLASS_OF_STATE_OF_PERSON",
-                    "STATE_OF_FUNCTIONAL_SYSTEM_DOMESTIC_PROPERTY",
-                    "OCCUPIER_LOCATED_IN_PROPERTY_ROLE",
-                    "DOMESTIC_PROPERTY_THAT_IS_OCCUPIED_ROLE",
-                    "OCCUPANT_LOCATED_IN_VOLUME_ENCLOSED_BY_PROPERTY_ASSOCIATION"
-                    ));
+        final Map<String, Thing> entities = mcService.findByEntityNameInTransaction(
+                List.of("CLASS_OF_STATE_OF_PERSON", "STATE_OF_FUNCTIONAL_SYSTEM_DOMESTIC_PROPERTY",
+                        "OCCUPIER_LOCATED_IN_PROPERTY_ROLE", "DOMESTIC_PROPERTY_THAT_IS_OCCUPIED_ROLE",
+                        "OCCUPANT_LOCATED_IN_VOLUME_ENCLOSED_BY_PROPERTY_ASSOCIATION"));
 
         // Find the required classes, kinds, and roles.
-        final ClassOfStateOfPerson classOfStateOfPerson = (ClassOfStateOfPerson) entities.get("CLASS_OF_STATE_OF_PERSON");
-        final ClassOfStateOfFunctionalSystem classOfStateOfFunctionalSystemDomesticProperty =
-            (ClassOfStateOfFunctionalSystem) entities.get("STATE_OF_FUNCTIONAL_SYSTEM_DOMESTIC_PROPERTY");
+        final ClassOfStateOfPerson classOfStateOfPerson = (ClassOfStateOfPerson) entities
+                .get("CLASS_OF_STATE_OF_PERSON");
+        final ClassOfStateOfFunctionalSystem classOfStateOfFunctionalSystemDomesticProperty = (ClassOfStateOfFunctionalSystem) entities
+                .get("STATE_OF_FUNCTIONAL_SYSTEM_DOMESTIC_PROPERTY");
         final Role occupierOfPropertyRole = (Role) entities.get("OCCUPIER_LOCATED_IN_PROPERTY_ROLE");
         final Role domesticOccupantInPropertyRole = (Role) entities.get("DOMESTIC_PROPERTY_THAT_IS_OCCUPIED_ROLE");
-        final KindOfAssociation occupantInPropertyKindOfAssociation =
-            (KindOfAssociation) entities.get("OCCUPANT_LOCATED_IN_VOLUME_ENCLOSED_BY_PROPERTY_ASSOCIATION");
+        final KindOfAssociation occupantInPropertyKindOfAssociation = (KindOfAssociation) entities
+                .get("OCCUPANT_LOCATED_IN_VOLUME_ENCLOSED_BY_PROPERTY_ASSOCIATION");
 
         // Create DbCreateOperations to create the objects and their properties.
-        final var personState = ExampleCommonUtils.mkUserBaseIri();
-        final var houseState = ExampleCommonUtils.mkUserBaseIri();
-        final var personParticipant = ExampleCommonUtils.mkUserBaseIri();
-        final var houseParticipant = ExampleCommonUtils.mkUserBaseIri();
-        final var houseOccupiedAssociation = ExampleCommonUtils.mkUserBaseIri();
+        final IRI personState = new IRI(USER_BASE, uid());
```

Comment:
The `mkUserBaseUri()` helper was introduced to be more readable, and I believe it is more readable - maybe a static import and name change would have been even more readable.

Reply:
I prefer this approach because it demonstrates how IRIs are meant to be used (without relying on an external helper function) when building objects. I think this approach makes it clear in a single line that to build an IRI for the `personState` object, you invoke the IRI constructor with a given IRI namespace, and an ID for the object.